target/linux/ramips/dts/Telco-Electronics-X1.dtsi

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "telco-electronics,x1", "mediatek,mt7621-soc";
+	
+	aliases {
+		led-boot = &system_led;
+		led-failsafe = &system_led;
+		led-running = &system_led;
+		led-upgrade = &system_led;
+	};
+	
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 18 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		
+		system_led: system {
+			label = "x1:green:system";
+			gpio-export,direction_may_change = <1>;
+			gpios = <&gpio0 26 GPIO_ACTIVE_LOW>;
+		};
+		
+		modemOffline {
+			label = "x1:red:modem-offline";
+			gpio-export,direction_may_change = <1>;
+			gpios = <&gpio0 23 GPIO_ACTIVE_HIGH>;
+		};
+		
+		modem4g {
+			label = "x1:blue:modem-4g";
+			gpio-export,direction_may_change = <1>;
+			gpios = <&gpio0 24 GPIO_ACTIVE_LOW>;
+		};
+		
+		modem3g {
+			label = "x1:green:modem-3g";
+			gpio-export,direction_may_change = <1>;			
+			gpios = <&gpio0 25 GPIO_ACTIVE_LOW>;
+		};
+		
+		led1 {
+			label = "x1:green:1";
```

Comment:
led1? Is this a link LED indicator for the LAN-Port 1? If so, you could please rename the label to "x1:green:lan1" and add support for it by adding something like
```ucidef_set_led_switch "lan1" "lan1" "$boardname:green:lan1" "switch0" "0x1"```
(or 0x2, 0x4, 0x8, 0x10 - whatever the mapping of the LAN to LED is) to the 01_leds?. Thanks.

Reply:
It is an array of LED's.  In the product they are used for LTE signal quality, but they could be used for anything.  I'll grep around for some examples